doc/classes/AspectRatioContainer.xml

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="AspectRatioContainer" inherits="Container" version="4.0">
+	<brief_description>
+		Aspect ratio container.
+	</brief_description>
+	<description>
+		Arranges controls in a way to keep the child's aspect ratio automatically whenever the container is resized. Solves the problem where container size is dynamic and the content size needs to adjust accordingly without loosing proportions.
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+	</methods>
+	<members>
+		<member name="alignment_horizontal" type="int" setter="set_alignment_horizontal" getter="get_alignment_horizontal" enum="AspectRatioContainer.AlignMode" default="1">
+			Specifies the relative position of child horizontally.
+		</member>
+		<member name="alignment_vertical" type="int" setter="set_alignment_vertical" getter="get_alignment_vertical" enum="AspectRatioContainer.AlignMode" default="1">
+			Specifies the relative position of child vertically.
+		</member>
+		<member name="ratio" type="float" setter="set_ratio" getter="get_ratio" default="1.0">
+			The aspect ratio to enforce on child. This is the width divided by the height. The ratio depends on the [member stretch_mode].
+		</member>
+		<member name="stretch_mode" type="int" setter="set_stretch_mode" getter="get_stretch_mode" enum="AspectRatioContainer.StretchMode" default="2">
+			The stretch mode used to align child.
+		</member>
+	</members>
+	<constants>
+		<constant name="STRETCH_WIDTH_CONTROLS_HEIGHT" value="0" enum="StretchMode">
+			The height of child is automatically adjusted based on the width of the container.
+		</constant>
+		<constant name="STRETCH_HEIGHT_CONTROLS_WIDTH" value="1" enum="StretchMode">
+			The width of child is automatically adjusted based on the height of the container.
+		</constant>
+		<constant name="STRETCH_FIT" value="2" enum="StretchMode">
+			The width and height of child is automatically adjusted to make the rect fit inside the rect of the container while keeping the aspect ratio.
```

Comment:
```suggestion
			The width and height of child controls is automatically adjusted to make their rect fit inside the rect of the container while keeping the aspect ratio.
```
Not sure if "rect" is clear enough here (and would likely be hard to localize), but "boundary rectangle" is a bit heavy too.

Reply:
I think it would be "bounding rectangle". To avoid verbosity, I rephrased this to:

> The bounding rectangle of child controls is automatically adjusted to fit inside the container while keeping the aspect ratio.